### PR TITLE
fix compiler warnings and MinGW build

### PR DESCRIPTION
**Pull request by Jojo-Schmitz**

### CODE REVIEW

File: all.h [hunk 1]
```diff
@@ -218,7 +218,6 @@
    // Undefined problematic #def'd macros in Microsoft headers
    #undef STRING_NONE
    #undef small
-   #undef DELETE
```
Comment (Jojo-Schmitz) [inline all.h]: moved to where it is really needed

File: libmscore/scorediff.h [hunk 2]
```diff
@@ -36,7 +36,7 @@ enum class ItemType {
 //---------------------------------------------------------
 //   DiffType
 //---------------------------------------------------------
-
+#undef DELETE
```
Comment (Jojo-Schmitz) [inline libmscore/scorediff.h]: needed not just for MSVC, but also for MinGW, and as thirdparty/diff_match_oatch unconditionally `#undef`'s it , we're doing it here too

File: mscore/scorecmp/scorecmp.cpp [hunk 3]
```diff
@@ -240,7 +240,7 @@ void ScoreComparisonTool::updateDiff()
 
 Score* ScoreComparisonTool::openScoreVersion(const ScoreVersion& ver)
       {
-      Score* s;
+      Score* s = 0;
```
Comment (anatoly-os) [inline mscore/scorecmp/scorecmp.cpp]: `nullptr` is more effective.

Comment (Jojo-Schmitz) [reply mscore/scorecmp/scorecmp.cpp]: well, its the same thing to C/C++ and less to type. And on occasions where I did use nullptr it didn't pass review ;-)

Comment (Jojo-Schmitz) [reply mscore/scorecmp/scorecmp.cpp]: but as you want it: here we go

File: libmscore/scorediff.h (hunk 2 above)
Comment (Jojo-Schmitz) [reply libmscore/scorediff.h]: It is not just me stumbling accross that, see https://musescore.org/en/node/277660

**Pull request merged**